### ISSUE TITLE
Support Box

### ISSUE THREAD

**Issue by latot** [r-lib/lintr#1087]
Hi, I'm using lintr in code and github, I think would be great support the box module, because actually lintr don't detect the functions contained in the modules.

https://github.com/klmr/box

Thx.

**Comment by AshesITR** [comment]
Hi, can you describe what kind of support you'd like?
Note that the `object_usage_linter` can see functions defined in your globalenv, so it may help to import your boxes before calling `lint()` if you see too many false positives from that particular linter.

**Comment by latot** [comment]
mm, I have a code, but seems the problem is in other place, i'll check it deeper, for now, I think would be great a fix for this:

```
b.R:1:11: style: Put spaces around all infix operators.
box::use(./a)
```

Thx.

**Comment by MichaelChirico** [comment]
seems like an unfortunate syntax choice by box to save 2 characters IMO

**Comment by AshesITR** [comment]
If you are using the development version of `lintr`, you could wrap your `box::use()` statements in a nolint block like this:

```
# nolint start: infix_spaces_linter.
box::use(./a)
# nolint end
```

**Comment by AshesITR** [comment]
In lintr 2.0.1, you can use `# nolint start` instead of `# nolint start: infix_spaces_linter.` as a workaround (though this will suppress all lints in the surrounded region, so use with care).

**Comment by AshesITR** [comment]
Yes, @klmr, the `infix_spaces_linter` uses the XML parse tree, so can do arbitrary XPath based rules to lint or not lint something.
To avoid false negatives, it's probably best to exactly match `box::use(<expr>/<expr>)` as a negative in this case.

Feel free to create a PR adding this exception to infix_spaces_linter.
